freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/krb/KrbKeytabGenerator.java

```diff
@@ -0,0 +1,70 @@
+package com.sequenceiq.freeipa.service.freeipa.user.krb;
+
+import com.cloudera.thunderhead.service.usermanagement.UserManagementProto.ActorKerberosKey;
+import com.sequenceiq.cloudbreak.common.service.Clock;
+import org.apache.kerby.kerberos.kerb.keytab.Keytab;
+import org.apache.kerby.kerberos.kerb.keytab.KeytabEntry;
+import org.apache.kerby.kerberos.kerb.type.KerberosTime;
+import org.apache.kerby.kerberos.kerb.type.base.EncryptionKey;
+import org.apache.kerby.kerberos.kerb.type.base.EncryptionType;
+import org.apache.kerby.kerberos.kerb.type.base.PrincipalName;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import javax.inject.Inject;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.Base64;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Kerberos keytab generator - This class is repsonsible for generating a keytab using existing keys
+ * belonging to an actor in User Management Service.
+ */
+@Component
+public final class KrbKeytabGenerator {
```

Comment:
Not a big deal, but could you drop `krb` in favor of `kerberos`? I think the package should be `user.kerberos` and the class name `UserKeytabGenerator`. This would reflect more of it's purpose in my opinion.

Reply:
Done.